source/common/network/connection_impl.cc

```diff
@@ -107,7 +107,7 @@ void ConnectionImpl::close(ConnectionCloseType type) {
     if (data_to_write > 0) {
       // We aren't going to wait to flush, but try to write as much as we can if there is pending
       // data.
-      transport_socket_->doWrite(*write_buffer_);
+      transport_socket_->doWrite(*write_buffer_, false);
```

Comment:
Yeah, I figured we were about to close so we don't need the extra syscall, and I thought I should comment that, but then didn't.  

I'll pass true, to make the code simpler/more-obvious, instead of trying the micro-optimization.